resources/js/api/cities.ts

```diff
@@ -0,0 +1,53 @@
+const API_BASE_URL = "/api/cities";
+
+function makeRequest<T>(
+  method: string,
+  url: string,
+  body: any = null,
+): Promise<T> {
+  return new Promise((resolve, reject) => {
+    const xhr = new XMLHttpRequest();
+    xhr.open(method, url, true);
+    xhr.setRequestHeader("Content-Type", "application/json");
+
+    xhr.onload = () => {
+      if (xhr.status >= 200 && xhr.status < 300 && xhr.status) {
+        if (xhr.status !== 204) resolve(JSON.parse(xhr.responseText) as T);
+        else resolve(null as T);
+      } else {
+        reject(JSON.parse(xhr.response));
+      }
+    };
+    xhr.onerror = () => reject(new Error("Network error occurred"));
+
+    xhr.send(body ? JSON.stringify(body) : null);
+  });
+}
+
+export function getCities(
+  url: string | null = null,
+  queryParams: string,
+): Promise<any> {
+  if (url) return makeRequest<any[]>("GET", url);
```

Comment:
I think it would be a great idea to create the entities types to get rid of these anys, that way we are going to have the function return typed.